### PR TITLE
Improve Integration-test to support testing to test with pods

### PR DESCRIPTION
**Pull request by gthao313**
support integration-test to be able to test with stateful set pods,
stateless nginx pods, and pods with PodDisruptionBudget.

**Issue number:**
#182 

**Description of changes:**
```
Author: Tianhao Geng <tianhg@amazon.com>
Date:   Thu Apr 7 00:15:02 2022 +0000

    Improve Integration-test to support testing to test with pods

    support integration-test to be able to test with stateful set pods,
    stateless nginx pods, and pods with PodDisruptionBudget.
```

**Testing done:**
Brupop test

```
NAME                          READY   STATUS    RESTARTS   AGE
nginx-test-5d59d67564-25rl2   1/1     Running   0          12m
nginx-test-5d59d67564-7jtx8   1/1     Running   0          12m
nginx-test-5d59d67564-9djrn   1/1     Running   0          17m
nginx-test-5d59d67564-czl7r   1/1     Running   0          17m
nginx-test-5d59d67564-ftf2h   1/1     Running   0          12m
web-test-0                    1/1     Running   0          17m
web-test-1                    1/1     Running   0          12m
web-test-2                    1/1     Running   0          11m
```
```
NAME                         MIN AVAILABLE   MAX UNAVAILABLE   ALLOWED DISRUPTIONS   AGE
pod-disruption-budget-test   2               N/A               6                     24m
```

```
NAME                                                STATE   VERSION   TARGET STATE   TARGET VERSION   CRASH COUNT
brs-ip-192-168-128-120.us-west-2.compute.internal   Idle    1.7.0     Idle                            0
brs-ip-192-168-152-67.us-west-2.compute.internal    Idle    1.7.0     Idle                            0
brs-ip-192-168-153-59.us-west-2.compute.internal    Idle    1.7.0     Idle                            0
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: integ/src/updater.rs [hunk 1]
```diff
@@ -158,6 +160,31 @@ pub async fn delete_cluster_resources(kube_config_path: &str) -> UpdaterResult<(
     Ok(())
 }
 
+// create/delete statefulset pods, stateless nginx pods, and pods with PDBs on EKS cluster
+pub async fn process_pods_test(action: &str, kube_config_path: &str) -> UpdaterResult<()> {
```
Comment (somnusfish) [inline integ/src/updater.rs]: Would it be better to make action an enum instead of string?

Comment (somnusfish) [inline integ/src/updater.rs]: Would it be better to make action an enum instead of string?

File: integ/src/main.rs [hunk 2]
```diff
@@ -204,9 +213,18 @@ async fn run() -> Result<()> {
                 .await
                 .context(error::TerminateEc2Instance)?;
 
+            // create different types' pods to test if brupop can handle them.
```
Comment (arnaldo2792) [inline integ/src/main.rs]: nit: Is this comment accurate?

File: integ/src/updater.rs [hunk 3]
```diff
@@ -158,6 +160,31 @@ pub async fn delete_cluster_resources(kube_config_path: &str) -> UpdaterResult<(
...
+    let pods_status = Command::new("kubectl")
```
Comment (arnaldo2792) [inline integ/src/updater.rs]: nit: I've seen that we usually refer to `<binary_name>` with the full path to the binary, also, this should be a constant.

File: integ/src/pods-template.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  clusterIP: None
+  selector:
+    app: nginx
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: web-test
+spec:
+  serviceName: "nginx"
```
Comment (arnaldo2792) [inline integ/src/pods-template.yaml]: nit: is this autogenerated? If not, remove the `"` to be consistent

File: integ/src/updater.rs [hunk 5]
```diff
@@ -15,6 +15,8 @@ use tokio_retry::{
 
 const BRUPOP_NODE_LABEL: &str = "bottlerocket.aws/updater-interface-version=2.0.0";
 const BRUPOP_NAMESPACE: &str = "brupop-bottlerocket-aws";
+const CURRENT_LOCATION_PATION: &str = "integ/src";
```
Comment (arnaldo2792) [inline integ/src/updater.rs]: nit: typo? `CURRENT_LOCATION_PATH`

**Review by cbgbt — APPROVED** [review]

**Review by somnusfish — APPROVED** [review]

**Pull request merged**